src/WebJobs.Script/Description/FunctionInvokerBase.cs

```diff
@@ -81,7 +86,80 @@ protected bool InitializeFileWatcherIfEnabled()
             return false;
         }
 
-        public abstract Task Invoke(object[] parameters);
+        public async Task Invoke(object[] parameters)
+        {
+            FunctionStartedEvent startedEvent = null;
+
+            // We require the ExecutionContext, so this will throw if one is not found.
+            ExecutionContext functionExecutionContext = (ExecutionContext)parameters.First(p => p?.GetType() == typeof(ExecutionContext));
```

Comment:
Any reason we need a specific type comparison here? Should this be `p is ExecutionContext` instead?